pkg/volume/util/volumepathhandler/volume_path_handler.go

```diff
@@ -113,28 +113,31 @@ func mapBindMountDevice(v VolumePathHandler, devicePath string, mapPath string,
 	linkPath := filepath.Join(mapPath, string(linkName))
 
 	file, err := os.Stat(linkPath)
-	if err != nil {
-		if !os.IsNotExist(err) {
-			return fmt.Errorf("failed to stat file %s: %v", linkPath, err)
-		}
 
-		// Create file
-		newFile, err := os.OpenFile(linkPath, os.O_CREATE|os.O_RDWR, 0750)
-		if err != nil {
-			return fmt.Errorf("failed to open file %s: %v", linkPath, err)
-		}
-		if err := newFile.Close(); err != nil {
-			return fmt.Errorf("failed to close file %s: %v", linkPath, err)
-		}
-	} else {
+	if err == nil {
 		// Check if device file
 		// TODO: Need to check if this device file is actually the expected bind mount
 		if file.Mode()&os.ModeDevice == os.ModeDevice {
 			klog.Warningf("Warning: Map skipped because bind mount already exist on the path: %v", linkPath)
 			return nil
 		}
+	} else {
+		if !os.IsNotExist(err) {
+			return fmt.Errorf("failed to stat file %s: %v", linkPath, err)
+		}
+	}
 
-		klog.Warningf("Warning: file %s is already exist but not mounted, skip creating file", linkPath)
+	if err := os.Remove(linkPath); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to remove file %s: %v", linkPath, err)
+	}
+
+	// Create file
+	newFile, err := os.OpenFile(linkPath, os.O_CREATE|os.O_RDWR, 0750)
```

Comment:
if linkPath already exist, will it return error?

Reply:
linkPath will be deleted before open operation, so the linkPath should not exist here. 
https://github.com/kubernetes/kubernetes/pull/102161/files/76defd36db916a65844b678f7bc4162940c777cb#diff-20bd7a839b6183eed4620321dd0da789bc8e823ac4839524e4fcd10b9cb5aabeR130